### PR TITLE
capability: fix CapabilityBoundingSet=~ does not work correctly.

### PR DESCRIPTION
**Pull request by iplayinsun**
There are one obvious bug, and other unclear code.

===================================================
One bug.

    CapabilityBoundingSet=~CAP_XXX_A
    CapabilityBoundingSet=~CAP_XXX_B

If you use CapabilityBoundingSet=~CAP_XXX as an even number as in the case above,
the result is (~CAP_XXX_A | ~CAP_XXX_B) in the code before the modification.
It means ~0, with is equivalent to the CAP_ALL value.

The expected value is (~CAP_XXX_A & ~CAP_XXX_B),
which is the same value as ~(CAP_XXX_A | CAP_XXX_B).

====================================================
Another unclear code

    CapabilityBoundingSet=~CAP_XXX_A
    CapabilityBoundingSet=~

As a result of "CapabilityBoundingSet=~", variable sum is 0 and variable invert is true.

And the sum becomes ~0, because of the "sum = invert ? ~sum : sum;" code.
So, "else { *capability_set |= sum;}" runs instead of "if (sum == 0".

Fortunately, the code "*capability_set |= sum;" also makes "~0".
But it is not clear.
The code "if (sum == 0)  { *capability_set = invert ? ~sum : sum; }"  is clearer.

In conclusion, the results are the same before or after the modification.
However, clear code is better to maintain.

### CODE REVIEW

**Comment by yuwata** [comment]
lgtm. If possible, could you update the commit message? I am sorry it is not clear for me...

**Comment by yuwata** [comment]
I do not catch the following. What do you want to say?
> Also, it does not logically clear the following items.
CapabilityBoundingSet=~

**Comment by iplayinsun** [comment]
Thanks review. 
I modify comment.

**Comment by yuwata** [comment]
Thank you for updating the explanation. I whould like to merge the commit with the following message.
```
core: merge the second CapabilityBoundingSet= lines by AND when it is prefixed with tilde

If a unit file contains multiple CapabilityBoundingSet= or AmbientCapabilities= lines, e.g.,
CapabilityBoundingSet=CAP_A CAP_B
CapabilityBoundingSet=~CAP_B CAP_C
before this commit, it results all capabilities except CAP_C are set to CapabilityBoundingSet=, as each lines are always merged by OR.
This commit makes lines prefixed with ~ are merged by AND. So, for the above example only CAP_A is set.
This makes easier to drop capabilities with drop-in config files.
```
What do you think?

**Comment by iplayinsun** [comment]
Good commit message.
Please merge with  this commit message.

Thanks very much.

**Comment by poettering** [comment]
any chance we can get a test for this added, too? feels like something brittle enough to deserve a test so that we don't regress on this any time soon!

**Comment by poettering** [comment]
(Ah ignore me, I see now that #6735 already has been posted with such a test added)
